hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -1301,6 +1327,46 @@ private int refreshCallQueue() throws IOException {
     return returnCode;
   }
 
+  /**
+   * Dumps the contents of the StateStore to stdout.
+   * @return true if it was successful
+   */
+  public static boolean dumpStateStore(Configuration conf,
+                                PrintStream output) throws IOException {
+    StateStoreService service = new StateStoreService();
+    conf.setBoolean(RBFConfigKeys.DFS_ROUTER_METRICS_ENABLE, false);
+    service.init(conf);
+    service.loadDriver();
+    if (!service.isDriverReady()) {
+      System.err.println("Can't initialize driver");
+      return false;
+    }
+    // Get the stores sorted by name
+    Map<String, RecordStore<? extends BaseRecord>> stores = new TreeMap<>();
+    for(RecordStore<? extends BaseRecord> store: service.getRecordStores()) {
+      stores.put(StateStoreUtils.getRecordName(store.getRecordClass()), store);
+    }
+    for (Entry<String, RecordStore<? extends BaseRecord>> pair: stores.entrySet()) {
+      output.println(String.format("---- %s ----", pair.getKey()));
+      if (pair.getValue() instanceof CachedRecordStore) {
+        for (Object record: ((CachedRecordStore) pair.getValue()).getCachedRecords()) {
+          if (record instanceof BaseRecord) {
+            BaseRecord baseRecord = (BaseRecord) record;
+            if (record instanceof PBRecord) {
```

Comment:
Can we just check for this right away? I think they instanceof was for baseRecord